src/Main.java

```diff
@@ -8,103 +8,63 @@ public static void main(String[] args) {
             System.out.println("Установите версию приложения для Android по ссылке");
         }
         //Task 2
+        clientOS = 0;
         int clientDeviceYear = 2015;
-        if (clientOS == 0) {
-            if (clientDeviceYear >= 2015) {
-                System.out.println("Установите версию приложения для iOS по ссылке");
-            } else {
+        if ((clientOS == 0) && (clientDeviceYear >= 2015)){
+            System.out.println("Установите версию приложения для iOS по ссылке");
+        } else if (clientDeviceYear < 2015) {
                 System.out.println("Установите облегченную версию приложения для iOS по ссылке");
-            }
-        } else if (clientOS == 1) {
-            if (clientDeviceYear >= 2015) {
-                System.out.println("Установите версию приложения для Android по ссылке");
-            } else {
-                System.out.println("Установите облегченную версию приложения для Android по ссылке");
-            }
+        } else if ((clientOS == 1) && (clientDeviceYear >= 2015)){
+            System.out.println("Установите версию приложения для Android по ссылке");
+        } else if (clientDeviceYear < 2015) {
+            System.out.println("Установите облегченную версию приложения для Android по ссылке");
         }
         //Task 3
         int year = 2021;
-        if ( (year < 1584) || ( (year % 4) != 0) || ( ( (year % 100) == 0)  && ( (year % 400) != 0) ) ) {
-            System.out.println("Год " + year + " - невисокосный");
-        } else {
+        if ( (year >= 1584) && ( ( (year % 4) == 0) &&  ( (year % 100) != 0) )  || ( (year % 400) == 0) )   {
```

Comment:
```suggestion
        if (year >= 1584 && year % 4 == 0 && year % 100 != 0  || year % 400 == 0)   {
```
скобки не нужны, много лишних ненужных скобок